app/components/organization-archive/list/index.ts

```diff
@@ -0,0 +1,127 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import type IntlService from 'ember-intl/services/intl';
+
+// eslint-disable-next-line ember/use-ember-data-rfc-395-imports
+import type DS from 'ember-data';
+import type Store from '@ember-data/store';
+
+import parseError from 'irene/utils/parse-error';
+import { type ArchiveListRef } from '..';
+import type OrganizationService from 'irene/services/organization';
+import type OrganizationArchiveModel from 'irene/models/organization-archive';
+
+interface OrganizationArchiveListSignature {
+  Args: {
+    ref: ArchiveListRef;
+  };
+}
+
+type OrganizationArchiveResponse =
+  DS.AdapterPopulatedRecordArray<OrganizationArchiveModel> & {
+    meta: { count: number };
+  };
+
+export default class OrganizationArchiveListComponent extends Component<OrganizationArchiveListSignature> {
+  @service declare organization: OrganizationService;
+  @service declare intl: IntlService;
+  @service declare store: Store;
+  @service('notifications') declare notify: NotificationService;
+
+  @tracked limit = 10;
+  @tracked offset = 0;
+  @tracked archiveResponse: OrganizationArchiveResponse | null = null;
+
+  constructor(owner: unknown, args: OrganizationArchiveListSignature['Args']) {
+    super(owner, args);
+
+    this.args.ref.reloadArchiveList = this.handleReloadArchiveList;
+
+    this.fetchArchives.perform(this.limit, this.offset);
+  }
+
+  get archiveList() {
+    return this.archiveResponse?.slice() || [];
+  }
+
+  get totalArchiveCount() {
+    return this.archiveResponse?.meta?.count || 0;
+  }
+
+  get hasNoArchive() {
+    return this.totalArchiveCount === 0;
+  }
+
+  get columns() {
+    return [
+      {
+        name: this.intl.t('organizationTableCreatedOn'),
+        component: 'organization-archive/list/created-on',
+      },
+      {
+        name: this.intl.t('organizationTableGeneratedBy'),
+        component: 'organization-archive/list/generated-by',
```

Comment:
Do we need a component for `generated-by` column? We can just the `valuePath` right?